src/mdns/mdns_mdnssd.cpp

```diff
@@ -1292,7 +1293,7 @@ void PublisherMDnsSd::ServiceInstanceResolution::HandleGetAddrInfoResult(DNSServ
     mInstanceInfo.mTtl = aTtl;
 
 exit:
-    if (!mInstanceInfo.mAddresses.empty() || aErrorCode != kDNSServiceErr_NoError)
+    if ((!mInstanceInfo.mAddresses.empty() && !moreComing) || aErrorCode != kDNSServiceErr_NoError)
```

Comment:
I'm thinking if we should remove this `!mInstanceInfo.mAddresses.empty()` check and just check the `!moreComing` flag? There are addresses dropped by the filter at line 1279 so maybe there aren't any addresses remaining while the `moreComing` flag is false.

Reply:
This PR's focus is on using `moreComing` flag (under mDNSResponder).

Removing `!mInstanceInfo.mAddresses.empty()` would be a bigger change. Basically saying if we think there are no (valid) addresses, we want to report it earlier (with an empty list), compared to current behavior where we would wait to see maybe new addresses are reported (wait for timeout).

Since this is a bigger change, I think may be good to be done in its own PR later.  I dont think it is good to add it in this PR. Always better to keep PRs smaller in scope and break changes into smaller PRs.

Also when adding this we should make sure all callbacks can properly handle when the `mAddresses` list is empty (exitsing code may have assumed that it would not be empty).